### ISSUE TITLE
Monitor/enable GitHub's secret detection feature

### ISSUE THREAD

**Issue by mogul** [ossf/allstar#363]
[GitHub now offers secret scanning alerts](https://www.bleepingcomputer.com/news/security/github-s-secret-scanning-alerts-now-available-for-all-public-repos/), but they have to be explicitly enabled.

This is exactly the kind of thing that one might want to detect/enforce via policy across a large set of repositories, so it seems like a great thing for Allstar to be able to manage!

**Comment by jeffmendoza** [comment]
Yes, if it has an API and is not easily done at the org level, we should have an Allstar policy to turn it on. Thanks for the suggestion!

**Comment by mogul** [comment]
[There's an API for configuring it per-repository.](https://docs.github.com/en/rest/secret-scanning?apiVersion=2022-11-28#about-secret-scanning) Nothing at the org level AFAICS.

**Comment by markdboyd** [comment]
I started looking into this and it seems like maybe it is possible to configure this at the org level?

https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/managing-security-and-analysis-settings-for-your-organization#enabling-or-disabling-a-feature-for-all-existing-repositories

**Comment by Big-Comfy** [comment]
I missed #813 earlier because I only checked open PRs.

Reading through that PR, I think a useful next pass is pretty concrete: keep the same small policy shape, add the README section, implement the `fix` action, and include a real run against a test repo in the PR body.

The one behavior I’d change from that branch is how a missing `security_and_analysis` block is reported. I’d rather not treat that exactly like “secret scanning is disabled,” since GitHub only returns that block when the app/user has the right visibility. That seems worth calling out separately.

I’d leave push protection out of the first patch unless maintainers want it bundled here too.

If that still sounds useful, I can put together a small PR against current `main`.
